tests/unit/test_scan.py

```diff
@@ -57,6 +58,10 @@ def test_should_log_error(self, mocker, caplog):
 
     class TestWhenAPISpecFileIsEmpty:
         def test_should_log_error(self, mocker, caplog):
+            mocker.patch(
+                "scanapi.scan.settings",
+                {"spec_path": "invalid_path/scanapi.yaml", "no_report": False},
```

Comment:
**I'm not so sure** but as far as I understood these `mocker.patch` included in the `test_scan.py` are necessary to avoid a `KeyError` when accessing the `no_report = settings["no_report"]` in the `scan` function. Maybe including a default value for the `no_report` option in the `Settings` class would solve this problem:

https://github.com/scanapi/scanapi/blob/ec9574034d976b120476f1eba585d83a28a8cee7/scanapi/settings.py#L12-L19

WDYT @JRubics @camilamaia?

Reply:
You are right that I forgot to add a default value to the Settings class, but that doesn't solve the problem you are referencing here.
As I understood, you need to add the necessary fields when mocking some object, because it doesn't use the real Settings class, it is just simulating it because scan needs the settings object. Because scan() uses only settings["spec_path"] and settings["no_report"], we should mock only those properties in the settings mock.
Please check it and correct me if I am wrong. I've updated the Settings class, so you only need to check the test_scan.py.

Reply:
Makes sense. My point is that with the default value, the tests will still pass, even if we remove the settings mock (not true for the first one). In this case, it will use the actual `Settings` class. But it's fair to keep the mocks since we are not testing the `Settings` class here, IMO. :+1: (I'm resolving the conversation, but feel free to reopen it if you want)